api/proto/teleport/discoveryconfig/v1/discoveryconfig.proto

```diff
@@ -80,3 +83,41 @@ enum DiscoveryConfigState {
   // DISCOVERY_CONFIG_STATE_SYNCING is used when the discovery process has started but didn't finished yet.
   DISCOVERY_CONFIG_STATE_SYNCING = 3;
 }
+
+// AWSEC2InstancesDiscovered represents an AWS EC2 Instance that was discovered.
+message AWSEC2InstancesDiscovered {
+  // Region is the AWS Region.
+  string region = 1;
+  // Integration is the AWS OIDC Integration name used to fetch the resource.
+  string integration = 2;
+  // InstanceID is the instance ID which uniquely identifies this resource in AWS.
+  string instance_id = 3;
+  // EnrollMode indicates the enroll mode used to auto-enroll this instance.
+  // Either using the EICE or the Script mode.
+  types.InstallParamEnrollMode enroll_mode = 4;
+  // EnrollStatus indicates whether the instance was successfully enrolled or if it failed.
+  AWSEC2EnrollmentStatus enroll_status = 5;
+  // EnrollMessage is an informational message used to further expand on the result of auto-enrolling this instance.
+  string enroll_message = 6;
```

Comment:
Where does the message come from? Can you give any examples? If we control the messages, perhaps it would be better to store the status in a more structured way and generate messages in the code. This way we could have more control over it.

Reply:
It's mostly used to return error messages.
In the future, it can also be used to warn the user about some not-ideal scenario (eg, missing optional IAM Policies)